### DISCUSSION TITLE
macOS runners: Are any homebrew packages required for normal operation?

### DISCUSSION THREAD

**Discussion by Lord-Kamina** [actions/virtual-environments#7505] [Installed software]
I'm using the macOS runner to install all dependencies of my project via homebrew. It's kinda slow, so I was looking into caching the homebrew folders and so far, so good.

Problem is, the images already come with a bunch of packages and thus the cache files are much larger than they really need to be.
In the spirit of optimizing my space consumption, I wanted to uninstall all the packages I don't need, prior to the cache generation.

So the question is, are there any homebrew packages that should **not** be uninstalled because they are actually needed by the runner itself to complete a run?

**Comment by ilia-shipitsin** [comment]
well. I'm afraid nobody knows.

however, runners are somewhat "ephemeral". it is no harm to delete something. the next run will start with brand new runner. no worry

**Reply by Lord-Kamina** [reply]
> well. I'm afraid nobody knows.
> 
> however, runners are somewhat "ephemeral". it is no harm to delete something. the next run will start with brand new runner. no worry

I know changes wouldn't persist, but I was still wondering if anything might impact the current run.


Anyway, for reference, here's what I've uninstalled with apparently no ill consecuences, so far:

```
          brew uninstall --force --formula ant apr apr-util argon2 aria2 aspell aws-sam-cli azure-cli bazelisk azure/bicep/bicep blackhole-2ch capstone carthage chromedriver colima composer firefox freetds gd geckodriver ghostscript gnupg go google-chrome gradle httpd hub imagemagick jasper jbig2dec jq julia kotlin krb5 libassuan libavif libeventt libidn libksba liblqr libpq libraw libslirp libssh libusb libxft libzip lima llvm@15 mad maven microsoft-edge mongodb/brew/mongodb-community@5.0 mongodb/brew/mongodb-database-tools mongosh  nginx node@18 npth oniguruma openjdk openjdk@17 opusfile p7zip packer php pinentry pipx postgresql@14 qemu r rustup-init sbt selenium-server session-manager-plugin six sox subversion swiftformat swig switchaudio-osx tcl-tk tidy-html5 unixodbc utf8proc vagrant vde virtualbox yq
```

There are several other that I thought might be uninstallable but turned out to be transitive dependencies of some of my dependencies, so I left them in.
